src/main/java/io/appform/dropwizard/sharding/dao/LookupDao.java

```diff
@@ -97,11 +102,13 @@ T getLockedForWrite(String lookupKey) {
          * @param lookupKey Id of the object
          * @return Extracted element or null if not found.
          */
-        T getLocked(String lookupKey, LockMode lockMode) {
-            return uniqueResult(currentSession()
-                    .createCriteria(entityClass)
-                    .add(Restrictions.eq(keyField.getName(), lookupKey))
-                    .setLockMode(lockMode));
+        T getLocked(String lookupKey, LockModeType lockMode) {
+            val session = currentSession();
+            CriteriaBuilder builder = session.getCriteriaBuilder();
+            CriteriaQuery<T> criteria = builder.createQuery(entityClass);
+            Root<T> root = criteria.from(entityClass);
+            criteria.where(builder.equal(root.get(keyField.getName()), lookupKey));
```

Comment:
This lack of finality of variables is fucking up my chi .. use `val` or designate with `final`

Reply:
Done